common/tags/src/main/java/org/apache/spark/annotation/InterfaceStability.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.annotation;
+
+import java.lang.annotation.Documented;
+
+/**
+ * Annotation to inform users of how much to rely on a particular package,
+ * class or method not changing over time.
```

Comment:
I probably should have said do we want to say something like classes not annotated are considered Unstable.  This would just cover if we missed something or clarify to the user if they found a class that wasn't annotated.  

It really only applies once we have the Audience tag and what those audience tags are.  You could have a tag that is for internal Spark developers/components (not applications using spark) which would apply to other components or components really closely tied to Spark (perhaps Bahir). In the Hadoop world some things are considered limited private (which I don't think we should use that tag) but are allowed to use some interfaces end user is not.

Not a big deal at this point.